### PR TITLE
Randomly spawned target per creature

### PR DESCRIPTION
**Pull request by Kisa75**

### CODE REVIEW

File: Assets/Scripts/MLAgents/Agents/GenericAgent.cs [hunk 1]
```diff
@@ -91,6 +90,11 @@ public override void Initialize()
         SetWalkerOnGround();
     }
 
+    public void SetTarget(Transform target)
```
Comment (nconotor) [inline Assets/Scripts/MLAgents/Agents/GenericAgent.cs:93]: Please use properties

File: Assets/Scripts/WalkTargetScript.cs [hunk 2]
```diff
@@ -109,10 +110,8 @@ public void FixedUpdate()
     /// </summary>
     /// <returns></returns>
     public void PlaceTargetCubeRandomly(){
-        var x = UnityEngine.Random.Range(4 , DynamicEnvironmentGenerator.TerrainSize - 4);
-        var z = UnityEngine.Random.Range(4, DynamicEnvironmentGenerator.TerrainSize  - 4);
-        //var y = TerrainGenerator.GetTerrainHeight(new Vector3(x, 0, z)) + + transform.localScale.y/2;
-        transform.localPosition = new Vector3(x, 0, z);
+        var newPos = terrainGenerator.miscTerrainData.SpawnPoints[UnityEngine.Random.Range(0, terrainGenerator.miscTerrainData.SpawnPoints.Count)].Item1;
```
Comment (nconotor) [inline Assets/Scripts/WalkTargetScript.cs:113]: Please use direct assignment

**Review by nconotor — CHANGES_REQUESTED** [review]

File: Assets/Scripts/MLAgents/Agents/GenericAgent.cs (hunk 1 above)
Comment (Kisa75) [reply Assets/Scripts/MLAgents/Agents/GenericAgent.cs:93]: No

File: Assets/Scripts/WalkTargetScript.cs (hunk 2 above)
Comment (Kisa75) [reply Assets/Scripts/WalkTargetScript.cs:113]: No

**Pull request merged**